presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -465,14 +465,14 @@ public SemiTransactionalHiveMetastore getMetastore()
     @Override
     public List<String> listSchemaNames(ConnectorSession session)
     {
-        return metastore.getAllDatabases(new MetastoreContext(session.getIdentity(), session.getQueryId(), session.getClientInfo(), session.getSource()));
+        return metastore.getAllDatabases(new MetastoreContext(session.getIdentity(), session.getQueryId(), session.getClientInfo(), session.getSource(), HiveUtil.getMetastoreHeader(session)));
```

Comment:
you can directly call the getMetastoreHeaders() in HiveSessionProperties

Reply:
✅